src/registrar/models/domain.py

```diff
@@ -279,55 +282,173 @@ def _create_host(self, host, addrs):
             logger.error("Error _create_host, code was %s error was %s" % (e.code, e))
             return e.code
 
+    def _convert_list_to_dict(self, listToConvert: list[tuple[str]]):
+        newDict = {}
+
+        # TODO-848: If duplicated nameserver names, throw error
+        for tup in listToConvert:
+            if len(tup) == 1:
+                newDict[tup[0]] = None
+            else:
+                newDict[tup[0]] = tup[1]
+        return newDict
+
+    def isSubdomain(self, nameserver):
+        return nameserver.find(self.name) != -1
+
+    def checkHostIPCombo(self, nameserver: str, ip: list):
+        if self.isSubdomain(nameserver) and (ip is None or ip == []):
+            raise ValueError("Nameserver %s needs to have an ip address" % nameserver)
+        elif not self.isSubdomain(nameserver) and (ip is not None and ip != []):
+            raise ValueError(
+                "Nameserver %s cannot be linked "
+                "because %s is not a subdomain" % (nameserver, ip)
+            )
+        return None
+
+    def getNameserverChanges(self, hosts: list[tuple[str]]):
+        """
+        calls self.nameserver, it should pull from cache but may result
+        in an epp call
+        returns tuple of four values as follows:
```

Comment:
**comment**: I think it should be possible to type hint the return type of this as `-> tuple[list, list, dict, list]` as an additional help alongside the comment.